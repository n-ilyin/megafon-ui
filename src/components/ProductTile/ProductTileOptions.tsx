import * as React from 'react';
import { cnCreate } from '../../utils/cn';
import './style/ProductTileOptions.less';
import BubbleHint from '../BubbleHint/BubbleHint';
import { IProductTileOptions } from './types';
import { productTileOptionsProps } from './propTypes';

const cn = cnCreate('mfui-product-tile-options');
class ProductTileOptions extends React.Component<IProductTileOptions> {
    static propTypes = productTileOptionsProps;

    renderFootnote(title: string, footnote: string): JSX.Element {
        const { onClickBubble } = this.props;

        return (
            <div className={cn('title')}>
                {title}
                <BubbleHint
                    className={cn('cashback-c')}
                    popupWidth="small"
                    placement="bottom"
                    click
                    trigger={<span className={cn('bubble-trigger')} onClick={onClickBubble}>Подробнее</span>}
                >
                    <div
                        className={cn('hint-text')}
                        dangerouslySetInnerHTML={{ __html: footnote }}
                    />
                </BubbleHint>
            </div>
        );
    }

    renderContent(title: string, caption: string, value: string, unit: string): JSX.Element {
        return (
            <div className={cn('content')}>
                <div className={cn('title')}>
                    {title}
                </div>
                <div className={cn('description')}>
                    {caption || `${value} ${unit}`}
                </div>
            </div>
        );
    }

    renderIcon(svgIcon: JSX.Element): JSX.Element {
        return (
            <div className={cn('icon')}>{svgIcon}</div>
        );
    }

    render() {
        const { options, head } = this.props;

        return (
            <div className={cn('')}>
                {!!head && <div className={cn('head')}>{head}</div>}
                <div className={cn('wrapper')}>
                    {options.map((option, index) => {
                        const { title, caption, value, unit, svgIcon, footnote } = option;

                        return (
                            <div className={cn('option')} key={title! + index}>
                                {this.renderIcon(svgIcon!)}
                                {footnote
                                    ? this.renderFootnote(title!, footnote)
                                    : this.renderContent(title, caption!, value!, unit!)
                                }
                            </div>
                        );
                    })}
                </div>
            </div>
        );
    }
}

export default ProductTileOptions;
