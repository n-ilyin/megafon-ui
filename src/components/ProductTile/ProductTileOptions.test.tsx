import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileOptions from './ProductTileOptions';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';

const props = {
    options: [{
        title: 'Безлимитные SMS',
        value: '55',
        unit: '₽ за 30 дней',
        svgIcon: <MegafonTv />,
    }],
};

describe('<ProductTileOptions />', () => {
    it('renders ProductTileOptions', () => {
        const wrapper = shallow(<ProductTileOptions {...props} onClickBubble={jest.fn()} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTileOptions with head', () => {
        const head = 'Дополнительные возможности:';
        const wrapper = shallow(<ProductTileOptions {...props} head={head} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTileOptions with caption', () => {
        const options = [{
            title: 'МегаФон ТВ',
            caption: 'Особый статус и привилегии',
            svgIcon: <MegafonTv />,
        }];
        const wrapper = shallow(<ProductTileOptions options={options} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders ProductTileOptions with footnote', () => {
        const options = [{
            title: 'Кэшбэк 20 %',
            footnote: 'Участвуйте в программе и тратьте кэшбэк',
            svgIcon: <MegafonTv />,
        }];
        const wrapper = shallow(<ProductTileOptions options={options} />);

        expect(wrapper).toMatchSnapshot();
    });
});
