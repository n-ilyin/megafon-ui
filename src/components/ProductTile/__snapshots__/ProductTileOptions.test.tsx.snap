// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileOptions /> renders ProductTileOptions 1`] = `
<div
  className="mfui-product-tile-options"
>
  <div
    className="mfui-product-tile-options__wrapper"
  >
    <div
      className="mfui-product-tile-options__option"
      key="Безлимитные SMS0"
    >
      <div
        className="mfui-product-tile-options__icon"
      >
        <test-file-stub />
      </div>
      <div
        className="mfui-product-tile-options__content"
      >
        <div
          className="mfui-product-tile-options__title"
        >
          Безлимитные SMS
        </div>
        <div
          className="mfui-product-tile-options__description"
        >
          55 ₽ за 30 дней
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductTileOptions /> renders ProductTileOptions with caption 1`] = `
<div
  className="mfui-product-tile-options"
>
  <div
    className="mfui-product-tile-options__wrapper"
  >
    <div
      className="mfui-product-tile-options__option"
      key="МегаФон ТВ0"
    >
      <div
        className="mfui-product-tile-options__icon"
      >
        <test-file-stub />
      </div>
      <div
        className="mfui-product-tile-options__content"
      >
        <div
          className="mfui-product-tile-options__title"
        >
          МегаФон ТВ
        </div>
        <div
          className="mfui-product-tile-options__description"
        >
          Особый статус и привилегии
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductTileOptions /> renders ProductTileOptions with footnote 1`] = `
<div
  className="mfui-product-tile-options"
>
  <div
    className="mfui-product-tile-options__wrapper"
  >
    <div
      className="mfui-product-tile-options__option"
      key="Кэшбэк 20 %0"
    >
      <div
        className="mfui-product-tile-options__icon"
      >
        <test-file-stub />
      </div>
      <div
        className="mfui-product-tile-options__title"
      >
        Кэшбэк 20 %
        <BubbleHint
          className="mfui-product-tile-options__cashback-c"
          click={true}
          placement="bottom"
          popupAlign="left"
          popupPadding="normal"
          popupWidth="small"
          trigger={
            <span
              className="mfui-product-tile-options__bubble-trigger"
            >
              Подробнее
            </span>
          }
        >
          <div
            className="mfui-product-tile-options__hint-text"
            dangerouslySetInnerHTML={
              Object {
                "__html": "Участвуйте в программе и тратьте кэшбэк",
              }
            }
          />
        </BubbleHint>
      </div>
    </div>
  </div>
</div>
`;

exports[`<ProductTileOptions /> renders ProductTileOptions with head 1`] = `
<div
  className="mfui-product-tile-options"
>
  <div
    className="mfui-product-tile-options__head"
  >
    Дополнительные возможности:
  </div>
  <div
    className="mfui-product-tile-options__wrapper"
  >
    <div
      className="mfui-product-tile-options__option"
      key="Безлимитные SMS0"
    >
      <div
        className="mfui-product-tile-options__icon"
      >
        <test-file-stub />
      </div>
      <div
        className="mfui-product-tile-options__content"
      >
        <div
          className="mfui-product-tile-options__title"
        >
          Безлимитные SMS
        </div>
        <div
          className="mfui-product-tile-options__description"
        >
          55 ₽ за 30 дней
        </div>
      </div>
    </div>
  </div>
</div>
`;
